### PR TITLE
Update MapTree's deep copying function to use a stack

### PR DESCRIPTION
**Pull request by noencke**
## Description

This ensures that very deep map trees will not stack overflow. This also moves the utility to a more central location and adds a unit test.

### CODE REVIEW

File: packages/dds/tree/src/test/simple-tree/mapTree.spec.ts [hunk 1]
```diff
@@ -1498,3 +1499,31 @@ describe("toMapTree", () => {
 		});
 	});
 });
+
+describe("mapTrees", () => {
+	it("can be deep copied", () => {
```
Comment (Josmithr) [inline packages/dds/tree/src/test/simple-tree/mapTree.spec.ts:1504]: Nit: When testing a specific function, I generally just name the test the same as the function. Makes it easier to discover IMO.

Comment (noencke) [reply packages/dds/tree/src/test/simple-tree/mapTree.spec.ts:1504]: Yea good idea

File: packages/dds/tree/src/core/tree/mapTree.ts [hunk 2]
```diff
@@ -30,3 +30,31 @@ export interface MapTree extends NodeData {
 export interface ExclusiveMapTree extends NodeData, MapTree {
 	fields: Map<FieldKey, ExclusiveMapTree[]>;
 }
+
+/**
+ * Returns a deep copy of the given {@link MapTree}.
+ */
+export function deepCopyMapTree(mapTree: MapTree): ExclusiveMapTree {
+	type Next = [fields: ExclusiveMapTree["fields"], sourceFields: MapTree["fields"]];
+	const rootFields = new Map();
```
Comment (Josmithr) [inline packages/dds/tree/src/core/tree/mapTree.ts:39]: Nit: stronger map typing?

File: packages/dds/tree/src/core/tree/mapTree.ts [hunk 3]
```diff
@@ -30,3 +30,31 @@ export interface MapTree extends NodeData {
 export interface ExclusiveMapTree extends NodeData, MapTree {
 	fields: Map<FieldKey, ExclusiveMapTree[]>;
 }
+
+/**
+ * Returns a deep copy of the given {@link MapTree}.
```
Comment (Josmithr) [inline packages/dds/tree/src/core/tree/mapTree.ts:35]: Probably worth a comment explaining why we implemented this iteratively.

**Review by Josmithr — APPROVED** [review]

**Pull request merged**